### ISSUE TITLE
Use "sendmail" or equivalent alias to send mail without SMTP

### ISSUE THREAD

**Issue by lbeltrame** [gogits/gogs#997]
This applies only if the SMTP server runs on the same host as Gogs. Use case: my mail server has a lot of checks for legitimate users, but OTOH this would require a huge burden for services (which usually do not need a full account). 

Therefore, using "sendmail" (quotes, because a lot of other MTAs like Postfix offer a sendmail-compatible interface) makes sense (of course, **only** if the system running Gogs and the mail server are one and the same) and doesn't even require a connection to the SMTP server.

**Comment by Unknwon** [comment]
Hi @lbeltrame , thanks for the feedback!

I'm not familiar with "sendmail" thing, would you mind to give some example? Or even better, a PR to show how code could be done?

**Comment by lbeltrame** [comment]
In data lunedì 02 marzo 2015 12:08:40, 无闻 ha scritto:

> I'm not familiar with "sendmail" thing, would you mind to give some example?
> Or even better, a PR to show how code could be done?

This is equivalent to the php mail() function, which invokes the "sendmail" 
binary (present on most Linux distros) which sends the mail either as local 
mail or using SMTP.

A very crude example I found online: 
http://www.snip2code.com/Snippet/172752/golang-mail%28%29--because-porting-awful-PHP

I can't do more than this as I have zero experience in Google....

**Comment by Unknwon** [comment]
Based on the link you give, it looks fairly easy, I think give a configure option would be good. How do you think?

**Comment by lbeltrame** [comment]
That would work, perhaps note in the doc that's Unix/Linux only.

**Comment by nd2s** [comment]
+1

**Comment by Naxiz** [comment]
+1

**Comment by tkisme** [comment]
+1

**Comment by iquidus** [comment]
+1

**Comment by hasufell** [comment]
yep, a full SMTP account is tedious if you want to have as little configuration as possible

**Comment by gnue** [comment]
+1

**Comment by renannprado** [comment]
+1

**Comment by halberl** [comment]
+1

**Comment by Jonny007-MKD** [comment]
Does gomail (@alexcesaro) have any ambitions to support the sendmail binary? Or is it only targeted for SMTP connections?

**Comment by alexcesaro** [comment]
You can easily use sendmail with Gomail. You just have to implement the [Sender](https://godoc.org/gopkg.in/gomail.v2#Sender) interface.

Just adapt the `mail` function of the link above to match `Sender.Send` signature and you're done.

**Comment by Jonny007-MKD** [comment]
Ok, thank you :) as soon as the other gomail PR is merged, I'll create one for this issue.

**Comment by lilmike** [comment]
Has there been any news on this issue? Really looking forward.
-Michael.

**Comment by jsieler** [comment]
Interested too. +1

**Comment by AlexDunmow** [comment]
+1

**Comment by MarcelSimon** [comment]
+1

**Comment by blankoworld** [comment]
Definitly need it as gomail seems not working very well here.

**Comment by JacobDrummond** [comment]
@Unknwon, has this feature requested attracted enough interest to be part of a milestone, or is it de-facto closed?

**Comment by blankoworld** [comment]
This is typically the kind of feature that makes Gitea have some users. Sens a mail is needed for our users to register. Sendmail-like is more useful than using private services.
As Gitea includes this fonctionnality, why not have a look into its code?

**Comment by Fastidious** [comment]
@blankoworld did you know that, if you have sendmail properly configured, you can use localhost/127.0.0.1 as the SMTP server?

**Comment by blankoworld** [comment]
That's the point: I didn't achieve this kind of configuration. I probably forget something. Have you any example of how it works: Gogs + sendmail + some external mail service (as Google, Gandi, etc)?

I finally install msmtp with only 10+ lines configuration. And link it to Gitea. Not because I prefer Gitea, but just because it works in a more easy way.

**Comment by MarcelSimon** [comment]
It is of course always possible to set up a local SMTP server. However, gogs is about easy configuration and easy deployment and I certainly would not consider setting up a SMTP server contributing to this goal. Proper and especially secure setup of SMTP is a discipline for itself imho. I can't imaging sendmail has by default a server running.

**Comment by plujon** [comment]
Here's a recpie for those who, like me, have postfix or a similar SMTP server running on the localhost:

    [mailer]
    ENABLED = true
    HOST = localhost:25
    FROM = "Gogs" <gogs@your.tld>
    SKIP_VERIFY = true

**Comment by webracer999** [comment]
+1 for sendmail support
